core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -481,6 +482,14 @@ public ActualProperties visitTableExecute(TableExecuteNode node, List<ActualProp
                     .build();
         }
 
+        @Override
+        public ActualProperties visitTableExecuteCoordinatorOnly(TableCoordinatorExecuteNode node, List<ActualProperties> inputProperties)
+        {
+            return ActualProperties.builder()
+                    .global(coordinatorSingleStreamPartition())
```

Comment:
Does it have to be run on coordinator?
Or can be any node? @findepi is ok to assume that every node can do arbitrary metadata/data operation with same permissions as coordinator?

Reply:
Per offline conversation. Yes it has to be on coordinator. As `ConnectorMetadata` is stateful and instantiated on coordinator only.

Reply:
I don't know for sure, I'd guess any node is good. Tough from when I started talking about this with @findepi we assumed it will run on coordinator node only.

Reply:
> Yes it has to be on coordinator. As ConnectorMetadata is stateful and instantiated on coordinator only.

let's add this as a code-level comment